test/import-index.js

```diff
@@ -0,0 +1,128 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/import-index';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	},
+	parserOptions: {
+		sourceType: 'module'
+	}
+});
+
+const error = {
+	ruleId: 'import-index',
+	message: 'Import index files with `.`'
+};
+
+ruleTester.run('import-index', rule, {
+	valid: [
+		`const m = require('.')`,
+		`const m = require('..')`,
+		`const m = require('../..')`,
+		`const m = require('./foobar')`,
+		`const m = require('foobar')`,
+		`const m = require('index')`,
+		`const m = require('index.js')`,
+		`const m = require('indexbar')`,
+		`const m = require('fooindex')`,
+		`const m = require('@index/foo')`,
+		`const m = require('@foo/index')`,
+		`const m = require('@foo/index.js')`,
+		`import m from '.'`,
+		`import m from '..'`,
+		`import m from '../..'`,
+		`import m from './foobar'`,
+		`import m from 'foobar'`,
+		`import m from 'index'`,
+		`import m from 'index.js'`,
+		`import m from 'indexbar'`,
+		`import m from 'fooindex'`,
+		`import m from '@index/foo'`,
+		`import m from '@foo/index'`,
+		`import m from '@foo/index.js'`
+	],
+	invalid: [
+		{
+			code: `const m = require('./')`,
+			errors: [error],
+			output: `const m = require('.')`
+		},
+		{
+			code: `const m = require('./index')`,
+			errors: [error],
+			output: `const m = require('.')`
+		},
+		{
+			code: `const m = require('./index.js')`,
+			errors: [error],
+			output: `const m = require('.')`
+		},
+		{
+			code: `const m = require('../../index.js')`,
+			errors: [error],
+			output: `const m = require('../..')`
+		},
+		{
+			code: `const m = require('./foo/index.js')`,
+			errors: [error],
+			output: `const m = require('./foo')`
+		},
+		{
+			code: `const m = require('./foobar/')`,
```

Comment:
Style-wise, my preference is to remove the trailing `/`, so this is :ok_hand: for me